src/os/user/iterate_windows.go

```diff
@@ -0,0 +1,77 @@
+package user
+
+import (
+	"internal/syscall/windows/registry"
+	"syscall"
+)
+
+// ProfileListKey registry key contains all local user/group SIDs as sub keys. It is a sub key of HKEY_LOCAL_MACHINE.
+// Since HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList registry key does not contain SIDs
+// of Administrator, Guest, or other default system users, some users or groups might not be provided when using
+// iterateUsers or iterateGroups
+const ProfileListKey = `SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList`
+
+// iterateSIDS iterates through ProfileListKey sub keys and calls provided fn with enumerated sub key
+// name as parameter. If fn returns non-nil error, iteration is terminated.
+func iterateSIDS(fn func(string) error) error {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, ProfileListKey, registry.QUERY_VALUE|registry.ENUMERATE_SUB_KEYS)
+	if err != nil {
+		return err
+	}
+	return k.ReadSubKeyNames(fn)
+}
+
+// iterateUsers iterates through ProfileListKey SIDs, looks up for user with each given SID and calls user provided fn
+// with each *User entry. Each iterated SID can be either user or group. Only user SIDs are processed.
+func iterateUsers(fn NextUserFunc) error {
+	return iterateSIDS(func(sid string) error {
+		SID, err := syscall.StringToSid(sid)
+
+		if err != nil {
+			return nil
+		}
+		// Skip non user SID
+		if _, _, accType, _ := SID.LookupAccount(""); accType != syscall.SidTypeUser {
+			return nil
+		}
+		u, err := newUserFromSid(SID)
+		if err != nil {
+			return nil
```

Comment:
return err?

Reply:
Most probably I mistakenly forgot these 2 cases. We should return nil only where we check if SID is not of User type.